sdk/head_tracker.cc

```diff
@@ -22,6 +22,65 @@
 #include "util/vectorutils.h"
 
 namespace cardboard {
+namespace {
+
+void GetViewporOrientationRotation(
+        const CardboardViewportOrientation ViewportOrientation,
+        Rotation* ekf_to_head_tracker,
+        Rotation* sensor_to_display) {
+
+  // This is the same than initializing the rotation from
+  // Rotation::FromYawPitchRoll(-M_PI / 2., 0, -M_PI / 2.).
+  static const Rotation kEkftoHeadTrackerLandscapeLeft =
+          Rotation::FromQuaternion(Rotation::QuaternionType(0.5, -0.5, -0.5, 0.5));
+  static const Rotation kSensorToDisplayLandscapeLeft =
+          Rotation::FromAxisAndAngle(Vector3(0, 0, 1), M_PI / 2.);
+
+  // This is the same than initializing the rotation from
+  // Rotation::FromYawPitchRoll(M_PI / 2., 0, M_PI / 2.).
+  static const Rotation kEkftoHeadTrackerLandscapeRight =
+          Rotation::FromQuaternion(Rotation::QuaternionType(0.5, 0.5, 0.5, 0.5));
+  static const Rotation kSensorToDisplayLandscapeRight =
+          Rotation::FromAxisAndAngle(Vector3(0, 0, 1), -M_PI / 2.);
+
+  // This is the same than initializing the rotation from
+  // Rotation::FromYawPitchRoll(M_PI / 2., M_PI / 2., M_PI / 2.).
+  static const Rotation kEkftoHeadTrackerPortrait =
+          Rotation::FromQuaternion(Rotation::QuaternionType(0.707107, 0, 0, 0.707107));
+  static const Rotation kSensorToDisplayPortrait =
+          Rotation::FromAxisAndAngle(Vector3(0, 0, 1), 0.);
+
+  // This is the same than initializing the rotation from
+  // Rotation::FromYawPitchRoll(-M_PI / 2., -M_PI / 2., -M_PI / 2.).
+  static const Rotation kEkftoHeadTrackerPortraitUpsideDown =
+          Rotation::FromQuaternion(Rotation::QuaternionType(0, -0.707107, -0.707107, 0));
+  static const Rotation kSensorToDisplayPortraitUpsideDown =
+          Rotation::FromAxisAndAngle(Vector3(0, 0, 1), M_PI);
+
+  switch (ViewportOrientation) {
+    case kCardboardViewportOrientationLandscapeLeft:
+      *ekf_to_head_tracker = kEkftoHeadTrackerLandscapeLeft;
+          *sensor_to_display = kSensorToDisplayLandscapeLeft;
+          break;
+
+    case kCardboardViewportOrientationLandscapeRight:
+      *ekf_to_head_tracker = kEkftoHeadTrackerLandscapeRight;
+          *sensor_to_display = kSensorToDisplayLandscapeRight;
+          break;
+
+    case kCardboardViewportOrientationPortrait:
+      *ekf_to_head_tracker = kEkftoHeadTrackerPortrait;
+          *sensor_to_display = kSensorToDisplayPortrait;
+          break;
+
+    default: // PortraitUpsideDown
```

Comment:
Consider creating a case for PortraitUpsideDown and leaving a default case with an empty body. This would be useful (and would minimize the refactoring) in case more cases are required in the future.